src/serverenvironment.cpp

```diff
@@ -1311,8 +1345,12 @@ void ServerEnvironment::step(float dtime)
 				block->setTimestampNoChangedFlag(m_game_time);
 
 				/* Handle ActiveBlockModifiers */
-				abmhandler.apply(block);
+				abmhandler.apply(block, abm_checked, abm_run, abm_not_cached);
 			}
+			g_profiler->avg("SEnv: active blocks", m_active_blocks.m_abm_list.size());
+			g_profiler->avg("SEnv: active blocks cached", m_active_blocks.m_abm_list.size() - abm_not_cached);
```

Comment:
Line is too long (> 80 characters)

Reply:
> Why do you need the cache at all? A boolean value would do it too ...

Huh? That's the whole point of this :)
ABMs are per node-type, not per block. We also do not want to invalidate all blocks somehow when a new ABM is added. The way it is now the cache is only invalidated when the block is changed and on top the decision is block local. That makes this simple and safe.

Reply:
Alright, I didn't even expect that the ABMs are processed from scratch each second - thought they were a static vector of ABM trigger information per content ID.

Reply:
@Cazadorro we can argue about the naming of the variables. :)
But please have a look at the code where they are used.

Reply:
@SmallJoker There're definitely more things to do. For example we could refresh the list of ABMs less frequently than we trigger them. In that case we could cache everything in a bit (as you suggest) until we refresh ABMs the next time.

Perhaps ABMs are not usually added often. So invalidating everything in that case might be OK. That would involve more complex logic with timestamps.